lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs

```diff
@@ -102,8 +102,28 @@ impl Client {
     /// It updates the local clipboard cache and returns the encoded message
     /// that should be sent to the RDP server.
     pub fn update_clipboard(&mut self, data: Vec<u8>) -> RdpResult<Vec<u8>> {
+        const CR: u8 = 0x13;
+        const LF: u8 = 0x10;
+
+        // convert LF to CRLF, as required by CF_OEMTEXT
+        let len_orig = data.len();
+        let mut converted = Vec::with_capacity(len_orig);
+        for i in 0..len_orig {
+            match data[i] {
+                LF if (i != 0) && data[i - 1] != CR => {
+                    converted.push(CR);
+                    converted.push(LF);
+                }
+                _ => converted.push(data[i]),
+            }
+            // Windows requires a null terminator, so add one if necessary
+            if i == len_orig - 1 && data[i] != 0x00 {
```

Comment:
nit: Should adding the null terminator be after the loop instead?

Reply:
I can't imagine it making a difference, but might as well move it since we know it only needs to happen once.